django/core/management/utils.py

```diff
@@ -173,3 +174,56 @@ def run_formatters(written_files, black_path=(sentinel := object())):
             [black_path, "--fast", "--", *written_files],
             capture_output=True,
         )
+
+
+SPACE_PREFIX = "    "
+BRANCH_PREFIX = "│   "
+TEE_PREFIX = "├── "
+ELBOW_PREFIX = "└── "
+
+
+def get_directory_tree(path, max_depth=10, prefix="", depth=0):
+    """
+    Get the folder structure of a directory and it's content,
+    akin to the GNU `tree` command.
+    Adapted from: https://stackoverflow.com/a/59109706
+
+    Example:
+    >>> for line in get_directory_tree(Path.home()):
+    ...     print(line)
+    """
+
+    path = Path(path)
+    if depth == 0:
+        yield colorize(path.name, fg="blue", opts=["bold"])
```

Comment:
Cool stuff! I’d be interested to hear why you settled on those styles. 

And this might need further work to support Django’s `--no-color` CLI flag, and `DJANGO_COLORS` environment variable, and checks for color support? See `django.core.management.color.color_style`.

Reply:
I actually tried to mimic the colors/style I got from `tree` as closely as possible. In hindsight, maybe my zsh config might have had a hand to play :P

I've made the changes to support user color config, the only hitch is supporting the `--no-color` flag. I really think `django.core.management.color.Style` should have a property/attribute `is_no_color`? It would make things a tad bit easier.